src/host/js/smooch.js

```diff
@@ -0,0 +1,93 @@
+import { VERSION } from '../../shared/js/constants/version';
+import hostStyles from '../stylesheets/iframe.less';
+import { waitForPage } from './utils/dom';
+import { init as initEnquire } from './utils/enquire';
+
+let iframe;
+const pendingOnCalls = [];
+let pendingInitCall;
+let pendingInitNext;
+let pendingInitCatch;
+
+
+const Smooch = {
+    VERSION,
+    on(...args) {
+        pendingOnCalls.push({
+            args
+        });
+    },
+    init(...args) {
+        pendingInitCall = args;
+        waitForPage(() => {
+            injectFrame();
+            hostStyles.use();
+        });
+        return {
+            then: (next) => pendingInitNext = next,
+            catch: (next) => pendingInitCatch = next
+        };
+    }
+};
+
+window.__onLibReady = function onSmoochReady(Lib) {
```

Comment:
yeah I'm even thinking about generating it randomly from the host and pass it to the iframe,